content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,260 @@ The following are not considered as part of the adversarial model:
 
-message ContentFilter {
-  string contentTopic = 1;
-}
+import "waku/message/v1/message.proto";
 
-message HistoryQuery {
-  // the first field is reserved for future use
-  string pubsubtopic = 2;
-  repeated ContentFilter contentFilters = 3;
-  PagingInfo pagingInfo = 4;
+message WakuMessageKeyValue {
+  optional bytes message_hash = 1; // Globally unique key for a Waku Message
+  optional waku.message.v1.WakuMessage message = 2; // Full message content as value
 }
 
-message HistoryResponse {
-  // the first field is reserved for future use
-  repeated WakuMessage messages = 2;
-  PagingInfo pagingInfo = 3;
-  enum Error {
-    NONE = 0;
-    INVALID_CURSOR = 1;
-  }
-  Error error = 4;
+message StoreRequest {
+  string request_id = 1;
+  bool return_values = 2; // Response should include full message content
+  
+  // Filter criteria for content-filtered queries
+  optional string pubsub_topic = 10;
+  repeated string content_topics = 11;
+  optional int64 time_start = 12;
+  optional int64 time_end = 13;
+
+  // List of key criteria for lookup queries
+  repeated bytes message_hashes = 20 // Message hashes (keys) to lookup
+  
+  // Pagination info. 50 Reserved
+  optional bytes pagination_cursor = 51; // Message hash (key) from where to start query (exclusive)
+  bool pagination_forward = 52;
+  optional uint64 pagination_limit = 53;
 }
 
-message HistoryRPC {
+message StoreResponse {
   string request_id = 1;
-  HistoryQuery query = 2;
-  HistoryResponse response = 3;
+
+  optional uint32 status_code = 10;
+  optional string status_desc = 11;
+
+  repeated WakuMessageKeyValue messages = 20;
+
+  optional bytes pagination_cursor = 51;
 }
 ```
+## General store concepts
 
-### Index
+### Waku message key-value pairs
 
-To perform pagination, 
-each `WakuMessage` stored at a node running the `13/WAKU2-STORE` protocol is associated with a unique `Index` that encapsulates the following parts. 
-- `digest`:  a sequence of bytes representing the SHA256 hash of a `WakuMessage`.
-  The hash is computed over the concatenation of `contentTopic` and `payload` fields of a `WakuMessage` (see [14/WAKU2-MESSAGE](/spec/14)).
-- `receiverTime`: the UNIX time in nanoseconds at which the `WakuMessage` is received by the receiving node.
-- `senderTime`: the UNIX time in nanoseconds at which the `WakuMessage` is generated by its sender.
-- `pubsubTopic`: the pubsub topic on which the `WakuMessage` is received.
+The store protocol operates as a query protocol for a key-value store of historical Waku messages,
+with each entry having a [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/) as value
+and [deterministic message hash](https://rfc.vac.dev/spec/14/#deterministic-message-hashing) as key.
+The store can be queried to return either a set of keys or a set of key-value pairs.
+Within the store protocol, Waku message keys and values MUST be represented in a `WakuMessageKeyValue` message.
+This message MUST contain the deterministic `message_hash` as key.
```

Comment:
```
The params proposed by @chaitanyaprem seems pretty complex, and the message hash based query don't need to depend on pubsub topic and content topics.

It seems we like making the query params splited based on scenarios, why not further use a different protocol id for history query and hash based query
```

Now that you mention and i give it a rethink, rather than making the query complex, we can just have 2 different RPC's in same Store protocol one for `messageHashQuery` and one for `TimeRange` based Query. That would simplify the protobuf as well. Anyways there is no rule that a protobuf serve only 1 RPC.

I don't think have different protocol ID's for each RPC is a good idea especially since both are different kind of queries (i.e Store Queries).